### DISCUSSION TITLE
Open in previous split

### DISCUSSION THREAD

**Discussion by PartyLich** [mcchrish/nnn.vim#114] [Q&A] [answered]
I recently updated to the latest version of the plugin, and suddenly found the following:

### expected behavior: 
 - File opens in the previously active split (vertical splits in this case)
 
### actual behavior:
- Files open only in the left split

### Reproduction:
- `nvim -u minimalvimrc`
- `:vs` to create a vertical split
- `c-w`  `l` (or whatever movement you prefer) to move to the right split
  ![image](https://user-images.githubusercontent.com/2792750/134551025-ee7abc88-6e27-48cd-948c-61b64ec39f37.png)
- `<leader> n` to open nnn
- select a file with `cr`
- The file opens in the left split, rather than the previously selected split
  ![image](https://user-images.githubusercontent.com/2792750/134551156-d2802038-92ed-40da-8f9c-cd530ea0c6f4.png)

### minimalvimrc
```
" minimalvimrc file
set nocompatible
set noswapfile

call plug#begin('~/.local/share/nvim/plugged')
Plug 'mcchrish/nnn.vim'
call plug#end()

" Nnn {{{
  " Opens the nnn window in a split
  let g:nnn#layout = 'new' " or vnew, tabnew etc.

  " Or pass a dictionary with window size
  let g:nnn#layout = { 'left': '~20%' }

  " Open nnn instead of netrw when opening a directory.
  let g:nnn#replace_netrw = 1

  " not actually used to reproduce the problem, but they were in the minimal config i tested with
  let g:nnn#action = {
      \ '<c-t>': 'tab split',
      \ '<c-s>': 'split',
      \ '<c-v>': 'vsplit',
      \ }
" }}}

```
nvim --version:
```
NVIM v0.5.0-dev+1155-g054eb31dc
Build type: RelWithDebInfo
```
`:PlugStatus`
![image](https://user-images.githubusercontent.com/2792750/134551620-ebc1d1e8-3306-48b9-b629-8656449c0bed.png)

Working back through tags, this change in behavior started with tag v3.1.  My assumption is that some configuration of `g:nnn#action` might enable me to return to the previous behavior (which required no config on tag v3.0.2 and earlier)....but I have no idea what that config value may be.

I'm guessing this may have been an intentional change, thus i'm writing this as a question and not a bug report. For me, at least, it *is* a breaking change. For now I'll just downgrade to the last tag that behaved the way I need it to, but I'd love some help with an actual solution.

EDIT to add: The note "Edit last selected file instead of the first" in the tag v3.4 release suggests maybe an attempt was made to address this behavior (or something very similar), but moving to that tag does not change the behavior I've described above.

**Comment by mcchrish** [comment]
You have two layout configs in the vimrc. Which one do you use?
```
  " Opens the nnn window in a split
  let g:nnn#layout = 'new' " or vnew, tabnew etc.

  " Or pass a dictionary with window size
  let g:nnn#layout = { 'left': '~20%' }
```
I assume you use the second  since it's the later one.

**Comment by mcchrish** [comment, marked as answer]
Can you try latest master if it fixes your issue?
https://github.com/mcchrish/nnn.vim/commit/ba64a4362844f19a0be61b54dbcccef378ce1c77

**Reply by PartyLich** [reply]
> I assume you use the second since it's the later one.
 
Yes. 

> ba64a43

This commit successfully fixes the behavior change noted above, in my brief testing thus far. Minimal config used below

```
" minimalvimrc file
set nocompatible
set noswapfile

call plug#begin('~/.local/share/nvim/plugged')
Plug 'mcchrish/nnn.vim', { 'commit': 'ba64a4362844f19a0be61b54dbcccef378ce1c77' }
call plug#end()

" Nnn {{{
  " Or pass a dictionary with window size
  let g:nnn#layout = { 'left': '~20%' }

  " Open nnn instead of netrw when opening a directory.
  let g:nnn#replace_netrw = 1
" }}}
```

Thank you!
